modules/integration/src/test/scala/scala/cli/integration/CompileTestDefinitions.scala

```diff
@@ -57,40 +87,43 @@ abstract class CompileTestDefinitions(val scalaVersionOpt: Option[String])
     }
   }
 
-  test("test scope") {
+  test("copy compile output") {
     val inputs = TestInputs(
       Seq(
-        os.rel / "Main.scala" ->
-          """//> using lib "com.lihaoyi::utest:0.7.10"
-            |
-            |object Main {
-            |  val err = utest.compileError("pprint.log(2)")
-            |  def message = "Hello from " + "tests"
-            |  def main(args: Array[String]): Unit = {
-            |    println(message)
-            |    println(err)
-            |  }
-            |}
-            |""".stripMargin,
-        os.rel / "Tests.scala" ->
-          """//> using lib "com.lihaoyi::pprint:0.6.6"
-            |//> using target.scope "test"
-            |
-            |import utest._
-            |
-            |object Tests extends TestSuite {
-            |  val tests = Tests {
-            |    test("message") {
-            |      assert(Main.message.startsWith("Hello"))
-            |    }
-            |  }
-            |}
-            |""".stripMargin
+        mainInput,
+        testInput
+      )
+    )
+    val tempOutput = os.temp.dir()
+    inputs.fromRoot { root =>
+      val outputOptions = Seq("--output", tempOutput.toString)
+      val cmd           = os.proc(TestUtil.cli, "compile", outputOptions, extraOptions, ".")
+      cmd.call(cwd = root, check = false, stderr = os.Pipe, mergeErrIntoOut = true)
+    }
+    val compileOutputCoppied = os.list(tempOutput).forall(_.baseName.startsWith("Main"))
```

Comment:
These two lines should be moved  inside the `inputs.fromRoot { … }`, so that they can use the output dir under `root`.

Reply:
We could also check for the actual file names, like
`os.list(tempOutput).map(_.relativeTo(tempOutput)) == Seq(os.rel / "Main.class", …)`

Reply:
let me know if it is ok now.